app/main.py

```diff
@@ -1,213 +1,57 @@
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    knights = []
+
+    for knight in knights_config.values():
+        knight_ = Knight(knight["name"], knight["power"], knight["hp"])
+        for armour in knight["armour"]:
+            armour_ = Armour(armour["part"], armour["protection"])
+            knight_.get_armour(armour_)
+
+        weapon = Weapon(knight["weapon"]["name"], knight["weapon"]["power"])
+        knight_.get_weapon(weapon)
+        if knight["potion"] is not None:
+            potion = Potion(knight["potion"]["name"],
+                            knight["potion"]["effect"])
+            knight_.get_potion(potion)
+        knights.append(knight_)
+
+    lancelot, arthur, mordred, red_knight = None, None, None, None
+
+    for knight in knights:
+        if knight.name == "Lancelot":
+            lancelot = knight
+        elif knight.name == "Arthur":
+            arthur = knight
+        elif knight.name == "Mordred":
+            mordred = knight
+        elif knight.name == "Red Knight":
+            red_knight = knight
```

Comment:
its bad practice
what if the knights have other names

Reply:
then what can i do ?